source/tests/unit/ni_fake_service_tests.cpp

```diff
@@ -1822,6 +1822,64 @@ TEST(NiFakeServiceTests, NiFakeService_AcceptViSessionArray_CallsAcceptViSession
   EXPECT_EQ(kDriverSuccess, response.status());
 }
 
+// Test for two-dimension mechanism
+TEST(NiFakeServiceTests, NiFakeService_UseTwoDimensionWithCorrectSize_CallsUseATwoDimensionParameter)
+{
+  nidevice_grpc::SessionRepository session_repository;
+  NiFakeMockLibrary library;
+  auto resource_repository = std::make_shared<FakeResourceRepository>(&session_repository);
+  nifake_grpc::NiFakeService service(&library, resource_repository);
+  std::uint32_t session_id = create_session(library, service, kTestViSession);
+  ViInt32 array[] = {1, 2, 2, 3, 3, 3};
+  ViInt32 array_lengths[] = {1, 2, 3};
+  ViInt32 array_size = 3;
+  // two-dimension call
+  EXPECT_CALL(library, UseATwoDimensionParameter(kTestViSession, _, _, array_size))
+      .WillOnce(Return(kDriverSuccess));
```

Comment:
Update this to test that the actual data is passed in.

There is a matcher called `ElementsAreArray` that is used in some tests. You still use the `_` for the param matcher, the array validation.

Reply:
Was able to update to use `ElementsAreArray` for the Sizes array but since we don't pass the total size of the array to the function we couldn't get it to work to check the array proper. @reckenro was able to help me create a new Matcher and called `MatchesArray` and use it in the function call in place of the discard.